tools/tests/source-maps.js

```diff
@@ -0,0 +1,69 @@
+var selftest = require('../selftest.js');
+var Sandbox = selftest.Sandbox;
+var files = require('../files.js');
+var catalog = require('../catalog.js');
+
+selftest.define("source maps from checkout", ['checkout'], function () {
```

Comment:
Sweet! I don't think this covers the case of throwing in a build plugin loaded into the tool, but @stubailo says he'll deal with that.